src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayViewModel.swift

```diff
@@ -0,0 +1,71 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import Combine
+
+class DiaryDayViewModel {
+
+	// MARK: - Init
+
+	init(day: DiaryDay, store: DiaryStoring) {
+		self.day = day
+		self.store = store
+
+		store.diaryDaysPublisher
+			.sink { [weak self] days in
+				guard let day = days.first(where: { $0.dateString == day.dateString }) else { return }
+
+				self?.day = day
+			}.store(in: &subscriptions)
+	}
+
+	// MARK: - Internal
+
+	@Published private(set) var day: DiaryDay
+	@Published var selectedEntryType: DiaryEntryType = .contactPerson
+
+	var entriesOfSelectedType: [DiaryEntry] {
+		day.entries.filter {
+			$0.type == selectedEntryType
+		}
+	}
+
+	func toggleSelection(of entry: DiaryEntry) {
+		entry.isSelected ? deselect(entry: entry) : select(entry: entry)
```

Comment:
Should `entry.isSelected` not toggle after selecting/deselecting the stuff?

Reply:
That is triggered by the store in the model layer and then comes back through the publisher in the correct selected/deselected state.